CRM/Contribute/BAO/Query.php

```diff
@@ -237,9 +237,6 @@ public static function whereClauseSingle(&$values, &$query) {
         return;
 
       case 'financial_type_id':
-        // @todo we need to make this resemble a hook approach.
-        CRM_Financial_BAO_FinancialType::getAvailableFinancialTypes($financialTypes);
-        $query->_where[$grouping][] = CRM_Contact_BAO_Query::buildClause("civicrm_contribution.$name", 'IN', array_keys($financialTypes), 'String');
```

Comment:
removing this line is the actual change - test shows it doesn't stop the ACL being applied - clause is being added twice